cognite/neat/session/_session/_data_model/_routes.py

```diff
@@ -16,30 +31,137 @@ def __init__(self, state: SessionState) -> None:
             return (
                 "<strong>No data model</strong>. Get started by reading data model with the <em>.read</em> attribute."
             )
 
         output = []
-
+        html = ""
         if self._state.data_model_store.provenance:
-            if self.physical:
-                html = self.physical._repr_html_()
-            if self.conceptual:
-                html = self.conceptual._repr_html_()
+            if physical := self.physical():
+                html += f"<H3>Physical</H3><br />{physical._repr_html_()} <br />"
+            if conceptual := self.conceptual():
+                html += f"<H3>Conceptual</H3><br />{conceptual._repr_html_()}"
             output.append(f"<H2>Data Model</H2><br />{html}")
 
         return "<br />".join(output)
+
+    def infer(
+        self,
+        model_id: dm.DataModelId | tuple[str, str, str] = (
+            "neat_space",
+            "NeatInferredDataModel",
+            "v1",
+        ),
+    ) -> IssueList:
+        """Data model inference from instances.
+
+        Args:
+            model_id: The ID of the inferred data model.
+
+        Example:
+            Infer a data model after reading a source file
+            ```python
+            # From an active NeatSession
+            ...
+            neat.read.xml.dexpi("url_or_path_to_dexpi_file")
+            neat.infer()
+            ```
+        """
+        self._state._raise_exception_if_condition_not_met("Data model inference", instances_required=True)
+        return self._infer_subclasses(model_id)
+
+    def _infer_subclasses(
+        self,
+        model_id: dm.DataModelId | tuple[str, str, str] = (
+            "neat_space",
+            "NeatInferredDataModel",
+            "v1",
+        ),
+    ) -> IssueList:
+        """Infer data model from instances."""
+        last_entity: DataModelEntity | None = None
+        if self._state.data_model_store.provenance:
+            last_entity = self._state.data_model_store.provenance[-1].target_entity
+
+        # Note that this importer behaves as a transformer in the data model store when there
+        # is an existing data model.
+        # We are essentially transforming the last entity's conceptual data model
+        # into a new conceptual data model.
+        importer = importers.SubclassInferenceImporter(
+            issue_list=IssueList(),
+            graph=self._state.instances.store.graph(),
+            data_model=last_entity.conceptual if last_entity is not None else None,
+            data_model_id=(dm.DataModelId.load(model_id) if last_entity is None else None),
+        )
+
+        def action() -> tuple[ConceptualDataModel, PhysicalDataModel | None]:
+            unverified_conceptual = importer.to_data_model()
+            unverified_conceptual = ToDMSCompliantEntities(rename_warning="raise").transform(unverified_conceptual)
+
+            extra_conceptual = VerifyConceptualDataModel().transform(unverified_conceptual)
```

Comment:
it's a nit and goes beyond current PR, but I belive these transformers better be named as with nouns and not berbs to signify a possible action. 
`ConceptualDataModelValidator`, `ConceptualDataModelMerger`, etc.

Currently they are named as if they were functions, not classes